File: pdlearning/src/microservices/course-microservice/Api/Application/Models/LearningPathCourseModel.cs
using System;
using Microservice.Course.Domain.Entities;

namespace Microservice.Course.Application.Models
{
    public class LearningPathCourseModel
    {
        public Guid Id { get; set; }

        public Guid CourseId { get; set; }

        public int? Order { get; set; }

        public static LearningPathCourseModel Create(LearningPathCourse entity)
        {
            return new LearningPathCourseModel
            {
                Id = entity.Id,
                CourseId = entity.CourseId,
                Order = entity.Order
            };
        }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/learning-path-course-model.ts
export interface ILearningPathCourseModel {
  id?: string;
  courseId: string;
  order: number;
}

export class LearningPathCourseModel implements ILearningPathCourseModel {
  public id?: string;
  public courseId: string;
  public order: number;
  constructor(data?: ILearningPathCourseModel) {
    if (data == null) {
      return;
    }
    this.id = data.id;
    this.courseId = data.courseId;
    this.order = data.order;
  }
}
